libs/coin-modules/coin-evm/src/logic/craftTransaction.ts

```diff
@@ -21,45 +19,37 @@ export async function craftTransaction(
     customFees?: FeeEstimation | undefined;
   },
 ): Promise<CraftedTransaction> {
-  const { amount, asset, recipient, sender, type } = transactionIntent;
+  const { sender } = transactionIntent;
+
+  const { type, to, data, value, gasLimit, feeData } = await prepareUnsignedTxParams(
+    currency,
+    transactionIntent,
+  );
 
-  const transactionType = getTransactionType(type);
-  const node = getNodeApi(currency);
-  const to = isNative(asset) ? recipient : (asset?.assetReference as string);
   const nonce = await getSequence(currency, sender);
-  const data = isNative(asset) ? Buffer.from([]) : getErc20Data(recipient, amount);
-  const value = isNative(asset) ? amount : 0n;
   const chainId = currency.ethereumLikeInfo?.chainId ?? 0;
-  const gasLimit = await node.getGasEstimation(
-    { currency, freshAddress: sender },
-    { amount: BigNumber(value.toString()), recipient: to, data },
-  );
-  const fee = await node.getFeeData(currency, { type: transactionType });
 
   const unsignedTransaction: TransactionLike = {
-    type: transactionType,
-    to,
+    type,
+    to: to,
     nonce,
     gasLimit: BigInt(gasLimit.toFixed(0)),
-    data: "0x" + data.toString("hex"),
-    value,
+    data: data,
+    value: value,
```

Comment:
More compact and coherent with `type` above.
```suggestion
    data,
    value,
```